### PR TITLE
Checkered cuticle: starter kits in search results

### PR DESCRIPTION
**Pull request by modernserf**
## Links
* Remix link: https://checkered-cuticle.glitch.me
* Manuscript link: https://glitch.manuscript.com/f/cases/3328157/
* Specs/Notion docs: https://www.notion.so/glitch/Search-244881f18e794507b67ef082a101cd84#0e64be9904114e8b9fd974e7def4ae4f

## GIF/Screenshots
Screen Shot 2019-04-05 at 3 37 01 PM
Screen Shot 2019-04-05 at 3 36 11 PM

## Changes:
* show starter kits on search page
* show starter kits in search autocomplete (behind Algolia feature flag)
* create StarterKitResult, MaskImage components & stories

## How To Test
* visit the search results page for a starter kit keyword (e.g. https://checkered-cuticle.glitch.me/search?q=react, https://checkered-cuticle.glitch.me/search?q=vr)
* turn on algolia search via https://checkered-cuticle.glitch.me/secret
* search in the top nav bar for starter kit keywords

## Feedback I'm looking for
* Do the starter kit results look right?
* Is it easier for marketing to manage the starter kits in this JS file, or would they prefer to work in algolia?
* Are the keywords / descriptions / cover colors acceptable?

### CODE REVIEW

File: stories/index.js [hunk 1]
```diff
@@ -370,6 +372,38 @@ storiesOf('SearchResults', module).add(
...
+    )),
+  );
+
+storiesOf('StarterKit', module).add('react', () => (
+  <StarterKitResult
+    result={{
+      id: 1,
+      type: 'starter-kit',
+      keywords: ['react'],
+      imageURL: 'https://glitch.com/culture/content/images/2018/10/react-starter-kit-1.jpg',
```
Comment (sarahzinger) [inline stories/index.js]: can I ask a stupid question? why don't we reference the like cdn image from the assets folder? Is it better to host it like this?

**Comment by sarahzinger** [comment]
**Very cool stuff! Some random thoughts:**

**On smaller/medium-ish screens I feel like the break points are different for the two images**
Screen Shot 2019-04-05 at 4 23 27 PM

**Should the two blocks here match in width? Seems like sometimes they do and sometimes they don't**
Screen Shot 2019-04-05 at 4 23 46 PM

**Was wondering how hard it is to add like a partial match, here are 2 instances for example where it seems like it'd be cool for it to pop up, but I don't know how hairy that would be implementation wise**
Screen Shot 2019-04-05 at 4 25 06 PM
Screen Shot 2019-04-05 at 4 25 34 PM

**Comment by modernserf** [comment]
Yeah, the sizes can be inconsistent, since its done with css grid's `minmax` instead of explicit break points. i'm not sure if anyone will notice if they're not doing the "Responsive Accordion"

File: src/components/search/starter-kit-result.styl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+.imageWrap
+  position: relative
+  height: 170px
+  max-width: 300px
+  margin-bottom: -70px
+  z-index: 1
+    
+.contentWrap
+  padding: 78px 8px 8px
```
Comment (ttseng) [inline src/components/search/starter-kit-result.styl]: padding should be `padding: 78px 12px 12px` to match other cards

File: src/components/search/starter-kit-result.styl [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  border-radius: 5px
+  h3
+    margin-top: 0
+  p
+    margin-bottom: 0
```
Comment (ttseng) [inline src/components/search/starter-kit-result.styl]: I noticed the standard font using text styling is using 16px font, but our default is 14px.  Can we edit that in `components/text/test.styl`?

File: src/state/search.js [hunk 4]
```diff
@@ -58,10 +58,49 @@ function formatCollection(hit) {
...
+    imageURL: 'https://glitch.com/culture/content/images/2018/10/react-starter-kit-1.jpg',
+    name: 'Build a Web App with React',
+    url: 'https://glitch.com/culture/react-starter-kit/',
+    description: 'A free, 5-part video course with interactive code examples that will help you learn React.',
+    coverColor: '#f0fcff',
+  },
+  {
+    id: 2,
+    type: 'starter-kit',
+    keywords: ['website'],
```
Comment (ttseng) [inline src/state/search.js]: I think we can safely expand the keywords here to something like 
`["website", "html", "css", "node", "javascript"]`

File: src/state/search.js [hunk 5]
```diff
@@ -58,10 +58,49 @@ function formatCollection(hit) {
...
+    imageURL: 'https://glitch.com/culture/content/images/2018/10/website-starter-kit-1.jpg',
+    name: 'How to Make a Website',
+    url: 'https://glitch.com/culture/website-starter-kit/',
+    description: 'A free, 4-part video course with interactive code examples to learn how to make a website using HTML, JS, CSS, and Node.js.',
+    coverColor: '#efe',
+  },
+  {
+    id: 3,
+    type: 'starter-kit',
+    keywords: ['vr', 'webvr'],
```
Comment (ttseng) [inline src/state/search.js]: suggest `['vr', 'webvr', 'a-frame']`

File: src/state/search.js [hunk 6]
```diff
@@ -58,10 +58,49 @@ function formatCollection(hit) {
   };
 }
 
-// top results
+// starter kits
+const starterKits = [
+  {
+    id: 1,
+    type: 'starter-kit',
+    keywords: ['react'],
```
Comment (ttseng) [inline src/state/search.js]: suggest `['react', 'redux'`]

File: src/state/search.js [hunk 7]
```diff
@@ -58,10 +58,49 @@ function formatCollection(hit) {
...
+    imageURL: 'https://glitch.com/culture/content/images/2018/10/react-starter-kit-1.jpg',
+    name: 'Build a Web App with React',
+    url: 'https://glitch.com/culture/react-starter-kit/',
+    description: 'A free, 5-part video course with interactive code examples that will help you learn React.',
+    coverColor: '#f0fcff',
```
Comment (ttseng) [inline src/state/search.js]: coverColor from when we previously featured this app in the homepage featured embed

coverColor: '#9de1ff'

File: src/state/search.js [hunk 8]
```diff
@@ -58,10 +58,49 @@ function formatCollection(hit) {
...
+  },
+  {
+    id: 2,
+    type: 'starter-kit',
+    keywords: ['website'],
+    imageURL: 'https://glitch.com/culture/content/images/2018/10/website-starter-kit-1.jpg',
+    name: 'How to Make a Website',
+    url: 'https://glitch.com/culture/website-starter-kit/',
+    description: 'A free, 4-part video course with interactive code examples to learn how to make a website using HTML, JS, CSS, and Node.js.',
+    coverColor: '#efe',
```
Comment (ttseng) [inline src/state/search.js]: coverColor: '#D0F1FF'

File: src/state/search.js [hunk 9]
```diff
@@ -58,10 +58,49 @@ function formatCollection(hit) {
...
+  },
+  {
+    id: 3,
+    type: 'starter-kit',
+    keywords: ['vr', 'webvr'],
+    imageURL: 'https://glitch.com/culture/content/images/2019/02/WebVR-Starter-Kit.-Part-1_-Intro-to-WebVR-1.png',
+    name: 'An Intro to WebVR',
+    url: 'https://glitch.com/culture/an-intro-to-webvr/',
+    description: 'A free, 5-part video course with interactive code examples that will teach you the fundamentals of WebVR using A-Frame.',
+    coverColor: '#fee',
```
Comment (ttseng) [inline src/state/search.js]: coverColor: '#9de1ff'

File: stories/index.js (hunk 1 above)
Comment (modernserf) [reply stories/index.js]: These images all come from the culture blog, they're not part of the ~community app

File: src/components/search/starter-kit-result.styl (hunk 3 above)
Comment (modernserf) [reply src/components/search/starter-kit-result.styl]: We should probably do that, but I'd need to verify that its not breaking anything else. I _also_ wanted to change how this component handles margins; maybe those can go together in the same ticket?

Comment (modernserf) [reply src/components/search/starter-kit-result.styl]: https://glitch.manuscript.com/f/cases/3328263/Text-component-margins-and-font-size-adjustments

**Pull request merged**